lib/ansible/template/__init__.py

```diff
@@ -1030,7 +1030,7 @@ def do_template(self, data, preserve_trailing_newlines=True, escape_backslashes=
             return res
         except (UndefinedError, AnsibleUndefinedVariable) as e:
             if fail_on_undefined:
-                raise AnsibleUndefinedVariable(e, orig_exc=e)
+                raise AnsibleUndefinedVariable(e)
```

Comment:
Tempted to separate these 2 and make it just a `raise` otherwise we lose the original context for the traceback (part of why orig_exc is there)

Reply:
```
except AnsibleUndefinedVariable:
   if fail_on_undefined:
       raise
...
except UndefinedError:
    if fail_on_undefined:
      raise AnsibleUndefinedVariable(e, orig_exc=e)
...
```